### PR TITLE
Update EOL versions (main)

### PR DESCRIPTION
**Pull request by richlander**

### CODE REVIEW

File: eng/pipelines/common/templates/pipeline-with-resources.yml [hunk 1]
```diff
@@ -79,9 +79,9 @@ extends:
         env:
           ROOTFS_DIR: /crossrootfs/x64
 
-      # We use a CentOS Stream 8 image here to test building from source on CentOS Stream 8.
+      # We use a CentOS Stream 8 image here to test building from source on CentOS Stream 9.
       SourceBuild_centos_x64:
-        image: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream8
+        image: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream9
```
Comment (carlossanlop) [inline eng/pipelines/common/templates/pipeline-with-resources.yml:84]: @dotnet/runtime-infrastructure someone please remind me - Do the eng/pipelines/common files come from arcade? If yes, then we cannot modify this file here directly, it needs to be modified in arcade. I just need confirmation.

Comment (jkoritzinsky) [reply eng/pipelines/common/templates/pipeline-with-resources.yml:84]: eng/pipelines/common are our files, not arcade's.

File: eng/pipelines/common/templates/pipeline-with-resources.yml [hunk 2]
```diff
@@ -79,9 +79,9 @@ extends:
         env:
           ROOTFS_DIR: /crossrootfs/x64
 
-      # We use a CentOS Stream 8 image here to test building from source on CentOS Stream 8.
+      # We use a CentOS Stream 8 image here to test building from source on CentOS Stream 9.
```
Comment (jkotas) [inline eng/pipelines/common/templates/pipeline-with-resources.yml:82]:
```suggestion
      # Test building from source on CentOS Stream.
```
Delete version numbers from the comment

File: eng/pipelines/runtime.yml [hunk 3]
```diff
@@ -1906,7 +1906,7 @@ extends:
         #
         # Sourcebuild legs
         # We have 3 important legs for source-build:
-        # - Centos.8 (ensures that known non-portable RID is working)
+        # - Centos.9 (ensures that known non-portable RID is working)
```
Comment (jkotas) [inline eng/pipelines/runtime.yml:1909]:
```suggestion
        # - Centos (ensures that known non-portable RID is working)
```

File: eng/pipelines/runtime.yml [hunk 4]
```diff
@@ -1906,7 +1906,7 @@ extends:
...
         # - Linux-x64 portable (used for dependency flow and downstream PR verification)
         # - Banana.24 - Non-existent RID to ensure we don't break RIDs we don't know about.
         #
```
Comment (jkotas) [inline eng/pipelines/runtime.yml:1912]: One more CentOS.8 reference below

File: eng/pipelines/common/templates/pipeline-with-resources.yml (hunk 1 above)
Comment (richlander) [reply eng/pipelines/common/templates/pipeline-with-resources.yml:84]: This also confused me initially.

**Comment by jkotas** [comment]
arm and arm64 linux musl images have a problem with Python installation:
```
ModuleNotFoundError: No module named 'requests'
+ exit 1
```

arm64 linux image has a problem with console encoding:
```
   at System.SR.get_Argument_EncodingNotSupported()
   at System.Text.EncodingTable.InternalGetCodePageFromName(System.String)
   at System.Text.EncodingTable.GetCodePageFromName(System.String)
   at System.Text.Encoding.GetEncoding(System.String)
   at System.Text.EncodingHelper.GetEncodingFromCharset()
   at System.ConsolePal.GetConsoleEncoding()
```

**Pull request closed without merging**